libraries/math/src/processor.rs

```diff
@@ -111,3 +111,120 @@ pub fn process_instruction(
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::instruction::MathInstruction;
+    use borsh::BorshSerialize;
+    use solana_program::account_info::AccountInfo;
+
+    #[test]
+    fn test_u64_multiply() {
+        assert_eq!(4, u64_multiply(2, 2));
+        assert_eq!(12, u64_multiply(4, 3));
+    }
+
+    #[test]
+    fn test_u64_divide() {
+        assert_eq!(1, u64_divide(2, 2));
+        assert_eq!(2, u64_divide(2, 1));
+    }
+
+    #[test]
+    fn test_f32_multiply() {
+        assert_eq!(4.0, f32_multiply(2.0, 2.0));
+        assert_eq!(12.0, f32_multiply(4.0, 3.0));
+    }
+
+    #[test]
+    fn test_f32_divide() {
+        assert_eq!(1.0, f32_divide(2.0, 2.0));
+        assert_eq!(2.0, f32_divide(2.0, 1.0));
+    }
+
+    #[allow(clippy::unit_cmp)]
+    #[test]
+    fn test_process_instruction() {
```

Comment:
This test would be more useful under `libraries/math/test` since it'll actually use the runtime

Reply:
@CriesofCarrots already told me that in some cases functional tests are more useful - I'll try to work on it. Right now I would like to see a better coverage in unit tests.

Reply:
Since this is testing the top-level `process_instruction` function and only-checking that it doesn't error, I would classify it as more of a top-level test.  The unit tests on the inner types, however, are definitely useful for providing more coverage on the math operations!

Reply:
I'll try to develop more useful things later :) Right now I want to get familiar with the code. I don't have a large experience in Rust testing. Rust has a great compiler, so it may be a little different situation here. For example in Symfony/PHP projects I like to have almost 100% code coverage in tests - even if you can not test result you can test if code don't produce any error at runtime. This may be a totally different situation here - different compilers :)

Reply:
Fine by me, we can refactor later then :smile: